documentation/source/examples/analytical_stein_thinning.rst

```diff
@@ -0,0 +1,169 @@
+Analytical example of Stein Thinning algorithm
+==============================================
+
+Step-by-step usage of the Stein Thinning algorithm (:cite:`liu2016kernelized`,
+:cite:`benard2023kernel`) on a small example with 10 data points in 2 dimensions
+and selecting a 2 point coreset, i.e., :math:`N=10, m=2`.
+
+For Stein Thinning, we need to provide a kernel and a score function, :math:`\nabla
+\log p(x)`, where :math:`p` is the underlying distribution of the data. In practice,
+we can estimate this using methods such as kernel density estimation and sliced score
+matching :cite:`song2020ssm`.
+
+In this example we will use ``PCIMQ`` kernel with ``length_scale`` of
+:math:`\frac{1}{\sqrt{2}}`: for two points :math:`x, y \in X`, :math:`k(x, y) =
+\frac{1}{\sqrt{1+\| x - y \|^2}}`. We use the standard normal density :math:`p(x)
+\propto e^{-\|x\|^2/2}`, so the score function is :math:`s_p(x) = -x`.
+
+The first step of the algorithm is to convert the given kernel into a Stein kernel
+using the given score function. The Stein kernel between points :math:`x` and
+:math:`y` is given as:
+
+.. math::
+    k_p(x,y) &= \langle \nabla_\mathbf{x}, \nabla_{\mathbf{y}} k(\mathbf{x},
+    \mathbf{y}) \rangle + \langle s_p(\mathbf{x}), \nabla_{\mathbf{y}} k(\mathbf{x}, \mathbf{y}) \rangle + \langle s_p(\mathbf{y}), \nabla_\mathbf{x} k(\mathbf{x}, \mathbf{y}) \rangle + \langle s_p(\mathbf{x}), s_p(\mathbf{y}) \rangle k(\mathbf{x}, \mathbf{y}) \\
+    & = -3\|x-y\|^2(1 + \|x-y\|^2)^{-5/2} + (d - \|x-y\|^2)(1 + \|x-y\|^2)
+    ^{-3/2} + (x\cdot y)(1 + \|x-y\|^2)^{-1/2}
+
+Now the algorithm proceeds iteratively, selecting the next point greedily by
+minimising the regularised KSD metric:
+
+.. math::
+    x_{t} = \arg\min_{x} \left( k_P(x, x)  + \Delta^+ \log p(x) -
+        \lambda t \log p(x) + 2\sum_{j=1}^{t-1} k_P(x, x_j) \right)
+
+Note that the Laplacian regularisation term (:math:`\Delta^+ \log p(x)`) vanishes for
+the given score function. We can now simplify the metric at iteration :math:`t` in
+this example:
+
+.. math::
```

Comment:
Can you just explain how one arrives at this (sub in `y=x` to the kernel and evaluate the log-density to ...